File: src/main/java/project/rpg/manager/ItemManager.java
package project.rpg.manager;

import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import project.rpg.items.ItemType;
import project.rpg.items.ManaRefillPotion;
import project.rpg.items.Wand;
import project.rpg.items.base.ItemBase;

import java.util.HashMap;
import java.util.Map;

public class ItemManager {

    private static final Map<ItemType, ItemBase> _itemMap = new HashMap<>();
    public  ItemManager(){

    }

    public static void makeItems(){

        _itemMap.put(ItemType.WAND, new Wand());
        _itemMap.put(ItemType.MANA_REFILLING_POTION, new ManaRefillPotion());
    }

    public static boolean isEquals(ItemStack item1, ItemStack item2){
        ItemMeta meta1 = item1.getItemMeta();
        ItemMeta meta2 = item2.getItemMeta();
        if(meta1.hasCustomModelData()&&meta2.hasCustomModelData())
            return meta1.getCustomModelData() == meta2.getCustomModelData();

        return false;
    }

    public static ItemStack getItem(ItemType itemType) {
        return _itemMap.get(itemType).getItem();
    }

}

File: src/main/java/project/rpg/manager/ItemManager.kt
package project.rpg.manager

import org.bukkit.inventory.ItemStack
import project.rpg.items.ItemType
import project.rpg.items.Items
import project.rpg.items.base.ItemBase
import project.rpg.items.objects.*

object ItemManager {
    //private val itemMap: MutableMap<Items, ItemBase> = EnumMap(Items::class.java)

    fun makeItems() {
        for (item in Items.values()) {
            item._item?.createItem()
        }
        //itemMap[Items.WAND] = wand
        //itemMap[Items.MANA_REFILLING_POTION] = manaRefillPotion
    }

    @JvmStatic
    fun getItem(value: Int): ItemBase? {
        for (item in Items.values()) {
            if (item.value==value) return item._item
        }
        return null
    }

    @JvmStatic
    fun getType(value: Int): ItemType? {
        for (item in Items.values()) {
            if (item.value==value) return item.type
        }
        return null
    }

    @JvmStatic
    fun isEquals(item1: ItemStack, item2: ItemStack): Boolean {
        val meta1 = item1.itemMeta
        val meta2 = item2.itemMeta
        return if (meta1.hasCustomModelData() && meta2.hasCustomModelData()) {
            meta1.customModelData == meta2.customModelData
        } else false
    }
}